src/main/java/org/terasology/logic/health/RegenAuthoritySystem.java

```diff
@@ -67,10 +74,14 @@
     public void update(float delta) {
         final long currentTime = time.getGameTimeInMs();
         // Execute regen schedule
-        invokeRegenOperations(currentTime);
+        if (currentTime > lastTick + CHECK_INTERVAL) {
+            invokeRegenOperations(currentTime);
+            lastTick = currentTime;
```

Comment:
_very_ minor point but to reduce the work done every frame you should do the addition of `CHECK_INTERVAL` here and not in the if condition